mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -251,13 +266,64 @@ struct UpstreamSettings {
     subscriptions: Vec<Direction>,
 }
 
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub enum StorageSettings {
+    #[serde(rename = "memory")]
+    Memory(MemorySettings),
+
+    #[serde(rename = "ring_buffer")]
+    RingBuffer(RingBufferSettings),
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct MemorySettings {
+    max_size: NonZeroUsize,
+}
+
+impl MemorySettings {
+    pub fn max_size(&self) -> NonZeroUsize {
+        self.max_size
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct RingBufferSettings {
+    max_file_size: NonZeroU64,
+    directory: PathBuf,
+    flush_options: FlushOptions,
+}
+
+impl RingBufferSettings {
+    pub fn max_file_size(&self) -> NonZeroU64 {
+        self.max_file_size
+    }
+
+    pub fn directory(&self) -> &PathBuf {
+        &self.directory
+    }
+
+    pub fn flush_options(&self) -> &FlushOptions {
+        &self.flush_options
+    }
+}
+
+impl Default for RingBufferSettings {
```

Comment:
iirc all default settings we read from default.json . can we do that similar way?

Reply:
Don't think we even need this, as if we go with the defaults.json option then these defaults for settings are useless right?

Reply:
Just removed the Default here as storage settings are a part of bridge settings that can use defaults.json